code/modules/mob/living/status_procs.dm

```diff
@@ -5,6 +5,11 @@
 
 ////////////////////////////// STUN ////////////////////////////////////
 
+/mob/living/proc/get_stun_resistance_multiplier()
+	if(HAS_TRAIT(src, TRAIT_STUNRESISTANCE))
+		return 0.5
+	return 1
```

Comment:
Make this a macro. Proc overhead bad.

Reply:
Proc overhead is a bad excuse to make this a macro due to the low number of times this will actually be called, making the performance cost of using a proc over a macro negligable.